plugin/src/main/java/org/opennms/velocloud/shell/connections/EditConnectionCommand.java

```diff
@@ -0,0 +1,92 @@
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+package org.opennms.velocloud.shell.connections;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.opennms.velocloud.client.api.VelocloudApiClientCredentials;
+import org.opennms.velocloud.client.api.VelocloudApiException;
+import org.opennms.velocloud.clients.ClientManager;
+import org.opennms.velocloud.connections.ConnectionManager;
+import org.opennms.velocloud.connections.ConnectionValidationError;
+
+import javax.ws.rs.ProcessingException;
+
+@Command(scope = "opennms-velocloud", name = "connection-edit", description = "Edit a connection", detailedDescription = "Edit an existing connection to a velocloud orchestrator")
+@Service
+public class EditConnectionCommand implements Action {
+
+    @Reference
+    private ConnectionManager connectionManager;
+
+    @Reference
+    private ClientManager clientManager;
+
+    @Option(name="-f", aliases="--force", description="Skip validation and save the connection as-is")
+    public boolean skipValidation = false;
+
+    @Argument(index = 0, name = "alias", description = "Alias", required = true, multiValued = false)
+    public String alias = null;
+
+    @Argument(index = 1, name = "url", description = "Orchestrator Url", required = true, multiValued = false)
+    public String url = null;
+
+    @Argument(index = 2, name = "apiKey", description = "Orchestrator API Key", required = true, multiValued = false, censor = true)
+    public String apiKey = null;
+
+    @Override
+    public Object execute() throws Exception {
+        if (!this.connectionManager.getAliases().contains(this.alias)) {
+            System.err.println(String.format("No connection with the given alias exists: %s",  this.alias));
+            return null;
+        }
+        if (!skipValidation) {
+            try {
+                clientManager.validate(VelocloudApiClientCredentials.builder()
+                        .withApiKey(apiKey)
+                        .withOrchestratorUrl(url)
+                        .build());
+            }
+            catch (ConnectionValidationError e) {
+                System.err.println(String.format("Failed to validate credentials: %s", e.getMessage()));
+                return null;
+            }
+        }
+        final var connection = this.connectionManager.getConnection(alias).get();
+        this.clientManager.purgeClient(connection.asVelocloudCredentials());
```

Comment:
This has potential for a race. I.e. when the client got purged, a collector runs and re-populates the cache and then the credentials are updated.

We should move the whole purging as an implementation detail to the `save` method and update the credentials before invalidating the cache. Maybe the cache event provides some kind of locking for this.